test/integration-test/src/tests/load.rs

```diff
@@ -192,16 +192,28 @@ fn pin_lifecycle() {
         let mut bpf = Bpf::load(bytes).unwrap();
         let prog: &mut Xdp = bpf.program_mut("pass").unwrap().try_into().unwrap();
         prog.load().unwrap();
+        prog.pin("/sys/fs/bpf/aya-xdp-test-prog").unwrap();
+    }
+
+    // should still be loaded since prog was pinned
+    assert_loaded!("pass", true);
+
+    // 2. Load program from bpffs
+    {
+        let mut prog = Xdp::from_pin("/sys/fs/bpf/aya-xdp-test-prog").unwrap();
```

Comment:
I think we're missing a test where a Xdp::from_pin program goes out of scope
without links and without being unpinned. We should assert that the program is
still loaded after.

Reply:
added, thanks.